engine/src/main/java/org/terasology/rendering/dag/SkyBandsNode.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.config.RenderingDebugConfig;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.cameras.Camera;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FrameBuffersManager;
+import org.terasology.rendering.world.WorldRenderer;
+
+import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.terasology.rendering.opengl.OpenGLUtils.*;
+
+
+/**
+ * TODO: Diagram of this node
+ * TODO: Separate this node into multiple SkyBandNode's
+ */
+public class SkyBandsNode implements Node {
+
+    @In
+    private Config config;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private FrameBuffersManager frameBuffersManager;
+
+    private RenderingConfig renderingConfig;
+    private Material blurShader;
+    private FBO sceneOpaque;
+    private FBO sceneSkyBand0;
+    private RenderingDebugConfig renderingDebugConfig;
+    private Camera playerCamera;
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        renderingDebugConfig = renderingConfig.getDebug();
+        blurShader = worldRenderer.getMaterial("engine:prog.blur");
+        playerCamera = worldRenderer.getActiveCamera();
+    }
+
+    @Override
+    public void process() {
+        enableWireframeIf(renderingDebugConfig.isWireframe());
```

Comment:
- The answer to the first question is yes.
- OpenGL state changes shouldn't be touched at this stage. I was very, very tempted of touching some in the SkyBands node but I decided not to suggest it so that we can continue focusing on wrapping things.
- PerformanceMonitor is not an opengl thing so I don't mind touching it. The content of every process() method of every node should really be bracketed by startActivity()/endActivity() as a rule.